test/e2e/mob_test.go

```diff
@@ -0,0 +1,46 @@
+package e2e_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/openshift/machine-config-operator/test/framework"
+	"github.com/openshift/machine-config-operator/test/helpers"
+	"github.com/stretchr/testify/require"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestOnClusterBuildLabel(t *testing.T) {
+	cs := framework.NewClientSet("")
+	ctx := context.TODO()
+
+	// Create and label a MachineConfigPool
+	deleteMCP := helpers.CreateMCP(t, cs, "on-cluster-build")
+
+	helpers.LabelRandomNodeFromPool(t, cs, "worker", "node-role.kubernetes.io/on-cluster-build")
```

Comment:
We can get away without having to label a random node, which should make the test execution a bit faster since we don't have to wait for the node to transition from one MachineConfig and then back to another at the end.